Basic debug interface setup for geolocation info.

Iryklaunavan opened the pull request:
There's a start/end button to track when a path "starts" and "ends" (when
the buttons are pressed), and a timer and coordinates are displayed. I don't have a mobile device that can access the Internet, and my laptop isn't really cooperating when I try roaming around campus with it, so if someone could upload it to a generic server and test out if the coordinates are updating more or less in real-time, that'd be great.

I left room for adding some results functionality, and the sending of
"path complete" data to wherever that needs to go. If there's anything
further you want me to do with this, let me know.

Also, I apparently accidentally included all the other files in the frontend clone in the commit. Not sure if that happens every time or if I messed up somewhere.

Iryklaunavan committed:
Basic debug interface setup for geolocation info.

There's a start/end button to track when a path "starts" and "ends" (when
the buttons are pressed), and a timer and coordinates are displayed. I don't have a mobile device that can access the Internet, and my laptop isn't really cooperating when I try roaming around campus with it, so if someone could upload it to a generic server and test out if the coordinates are updating, that'd be great.

I left room for adding some results functionality, and the sending of
"path complete" data to wherever that needs to go. If there's anything
further you want me to do with this, or if I misinterpreted something, let
me know.

bgw commented on StartEnd.html:
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
```
Try to follow the style of [demo.html](https://github.com/routemaster/routemaster-frontend/blob/master/src/html/demo.html). Indent your nested tags and don't put spaces around all the equals signs.

bgw commented on StartEnd.html:
```diff
@@ -0,0 +1,27 @@
+
+<p id = "Location" style = "display:none"></p>
+
+<p></p>
```
`<p>` tags should contain a body, not delimit. For example, this code should be:

```
<p>
    <button id="StartButton">Start</button>
</p>
```

bgw commented on StartEnd.html:
```diff
@@ -0,0 +1,27 @@
+
+<p></p>
+
+<button id = "StartButton">Start</button>
```
Buttons are typically associated with forms. I think the more semantically correct approach would be to use a `<a>` tags. (My code examples use `<button>` to match your code, but I think they should be `<a>` tags)

kalgynirae replied:
As far as I can tell, the correct solution here is to use `<button type="button">`. Without specifying the type, it defaults to having the semantics of a submit button, which is not what we want.

bgw replied:
@kalgynirae I concede. I just checked the W3c spec. [The button element](http://www.w3.org/html/wg/drafts/html/master/forms.html#the-button-element) is valid [in any phrasing context](http://www.w3.org/html/wg/drafts/html/master/dom.html#phrasing-content-1). What I'm confused about now is why so many people do CSS-buttons with `<a>` tags.

bgw commented on StartEnd.html:
```diff
@@ -0,0 +1,27 @@
+
+<p></p>
+
+<button id = "EndButton" disabled>End</button>
```
Instead of saying "disabled", try to use a CSS [class attribute](https://developer.mozilla.org/en-US/docs/HTML/Global_attributes#attr-class):

```
<button id="EndButton" class="disabled"></button>
```

bgw commented on StartEnd.html:
```diff
@@ -0,0 +1,27 @@
+<button id = "EndButton" disabled>End</button>
+
+</body>
+<script type = "text/javascript" src = "StartEnd.js">
```
This is debatable, but I think we're putting script tags in the `<head>` for this project.

bgw commented on StartEnd.js:
```diff
@@ -0,0 +1,89 @@
+(function() 
```
Use four-space indents for your code. Not tabs. You should be able to configure your editor to do this for you when you hit the tab key.

The body immediately after

```
(function() {
```

should be indented. Function brackets should be on the same line as function definitions.

bgw commented on StartEnd.js:
```diff
@@ -0,0 +1,89 @@
+	resultsbutton.style.display = "inline";
+	loctext.style.display = "none";
+	navigator.geolocation.clearWatch(watchPosId);
+	/*
```
You can insert single-line comments

```
// like this
```

bgw commented:
Most of my issues [are stylistic](https://github.com/routemaster/routemaster-frontend/wiki/Coding-Guidelines), but you also need to put your source in the `src/html` and `src/js` directory, and then build with maven.

The good news is that I can verify that the page does work as expected, but I'd like to get the stylistic issues fixed before we merge.

bgw commented:
Just to update the discussion, @kalgynirae cleaned up the code in the [debuginterface-debug](https://github.com/routemaster/routemaster-frontend/tree/debuginterface-debug) branch, and I'm now working on another (not yet pushed) branch to port the code to [Backbone.js](http://backbonejs.org/).

Iryklaunavan committed:
Apologies for the delay. Format should be better.

For some detail on what I had to swap. I needed the function to call
itself, which required I give it a self-identifier. The reason the
function needed to be able to call itself is because it needs to wait
until the document is loaded so that it can actually reference the buttons. It should still be anonymous outside of the context of the function itself, so I
don't think that will pose a significant issue.

That's the primary reason I placed the script after the body, initially.
The rest of the formatting stuff was just my being incompetent, and I
believe I fixed it according to the suggestions. I also kept the
timer/results dummy section, because I figured leaving it in would just be
simpler. We can simply remove it later if we choose not to relate them.

I'm also having some trouble with Maven, which is part of why I've waited
a while to upload what I have. If we have time I'll address it
tomorrow in class.

bgw commented:
I've added a [competing pull request here](https://github.com/routemaster/routemaster-frontend/pull/10).

kalgynirae commented:
We're merging #10 instead of this one.
